src/dataExplorer/context/resultsView.tsx

```diff
@@ -47,19 +65,67 @@ export const ResultsViewProvider: FC = ({children}) => {
     } as SimpleTableViewProperties,
   })
 
-  const [viewOptions, setViewOptions] = useSessionStorage(
+  // what can be chosen (e.g. the list of all options)
+  const [viewOptionsAll, saveViewOptionsAll] = useSessionStorage(
+    'dataExplorer.resultsOptions.all',
+    DEFAULT_VIEW_OPTIONS
+  )
+  const setViewOptionsAll = useCallback(
+    (updatedOptions: Partial<ViewOptions>) => {
+      saveViewOptionsAll({...viewOptionsAll, ...updatedOptions})
+    },
+    [viewOptionsAll]
+  )
+
+  // default options (a.k.a. based on schema)
+  const [defaultViewOptions, saveDefaultViewOptions] = useSessionStorage(
+    'dataExplorer.resultsOptions.default',
+    DEFAULT_VIEW_OPTIONS
+  )
+  const setDefaultViewOptions = useCallback(
+    (updatedOptions: Partial<ViewOptions>) => {
+      saveDefaultViewOptions({...defaultViewOptions, ...updatedOptions})
+    },
+    [defaultViewOptions]
+  )
+
+  // what was chosen (e.g. sublist chosen)
+  const [selectedViewOptions, saveSelectedViewOptions] = useSessionStorage(
     'dataExplorer.resultsOptions',
-    {}
+    DEFAULT_VIEW_OPTIONS
   )
+  const selectViewOptions = useCallback(
+    (updatedOptions: Partial<ViewOptions>) => {
+      saveSelectedViewOptions({...selectedViewOptions, ...updatedOptions})
+    },
+    [selectedViewOptions]
+  )
+
+  useEffect(() => {
+    // if parent query is re-run => decide what to reset in subquery viewOptions
+
+    // reset groupby
+    const excludeFromGroupby = NOT_PERMITTED_GROUPBY
+    const groupby = Object.keys(
+      resultFromParent?.parsed?.table?.columns || {}
+    ).filter(g => !excludeFromGroupby.includes(g))
```

Comment:
What does `g` represent here? Shouldn't this be `c`, since this is an array of column keys? In fact, `c` is a bit ambiguous as well. I think these single variables should be given full names to describe what it is these `filter` functions are doing

Reply:
Here it's definitely the columnName. Can make that explicit.